File: backend/entity/history.go
package entity

import (
	"time"

	"gorm.io/gorm"
)

type History struct {
	gorm.Model
	PlayedAt time.Time `json:"played_at"`
	SoundID  uint      `json:"sound_id"`
	Sound    *Sound    `gorm:"foreignKey:SoundID" json:"sound"`
	MemberID uint      `json:"member_id"`
	Member   *Member   `gorm:"foreignKey:MemberID" json:"member"`
}

File: frontend/src/interfaces/History.ts
import type { Sound } from "./Sound";
import type { Member } from "./Member";

export interface History {
  ID: number;
  played_at: string;
  sound_id: number;
  sound?: Sound;
  member_id: number;
  member?: Member;
}

export interface CreateHistoryRequest {
  played_at: string;
  sound_id: number;
  member_id: number;
}
